module-1/lab-functional-programming/your-code/Q1.ipynb

```diff
@@ -19,15 +19,15 @@
   },
```

Comment:
Good try to link concepts as list comprehension. When you double for loops, you should balance shorter code and readable code. The statement below would be more readable with a double for loop 
```Python

    bag_of_words = []

    [bag_of_words.append(words) for text in corpus for words in text.split() if words not in bag_of_words]
#It could be somthing like this 
    for s in corpus:
        s = ''.join(c for c in s if c not in punctuation)
        terms = s.split()
        for term in terms:
            if not term in bag_of_words and not term in stop_words:
                bag_of_words.append(term)
```